pkg/deploy/service_account.go

```diff
@@ -18,7 +18,12 @@ import (
 
 func SyncServiceAccountToCluster(deployContext *DeployContext, name string) (bool, error) {
 	saSpec := getServiceAccountSpec(deployContext, name)
-	return CreateIfNotExists(deployContext, saSpec)
+	isCreated, err := CreateIfNotExists(deployContext, saSpec)
+	if err != nil {
+		return false, err
+	}
+	// we're done if SA already exist
+	return !isCreated, nil
```

Comment:
```suggestion
	_, err := CreateIfNotExists(deployContext, saSpec)
	return err == nil, err
```

Reply:
well, it depends on policy whether we should do reconcile after object is created/modified or not.

Reply:
let's discuss it